src/bucket/BucketOutputIterator.cpp

```diff
@@ -157,9 +156,31 @@ BucketOutputIterator<BucketT>::put(typename BucketT::EntryT const& e)
         mBuf = std::make_unique<typename BucketT::EntryT>();
     }
 
+    if constexpr (std::is_same_v<BucketT, LiveBucket>)
```

Comment:
@SirTyson is this correct? Do we want to do this conversion on other BucketT types?

Reply:
Ya, the constexpr check is correct, since this only applies to `LiveBucket` types. We can still simplify this though. Try
```
// Comment about rewrite here
bool shouldRewrite = std::is_same_v<BucketT, LiveBucket> && mRewriteLiveToInitEntries && e.type() == LIVEENTRY &&
            protocolVersionStartsFrom(
                mMeta.ledgerVersion,
                BucketT::FIRST_PROTOCOL_SUPPORTING_INITENTRY_AND_METAENTRY);
if (shouldRewrite)
{
            ++mMergeCounters.mOutputIteratorLiveToInitRewrites;
            auto eCopy = e;
            eCopy.type(INITENTRY);
            *mBuf = eCopy;
}
else
{
            *mBuf = e;
}

++mMergeCounters.mOutputIteratorBufferUpdates;
```

Reply:
I need the block to be conditional on `constexpr (std::is_same_v<BucketT, LiveBucket>)` in order to compile, as other `BucketT::EntryT` have different types, no?

For example, when  `BucketT` is a `HotArchiveBucket`, `e.type() == LIVEENTRY` is a comparison of different enumeration types (`BucketEntryType` to `HotArchiveBucketEntryType`), and `eCopy.type(INITENTRY)` doesn't compile as there's no `HotArchiveBucketEntry::type(BucketEntryType t)`.

Reply:
Ahh, rip `constexpr`. This is fine thing, but I'd still like to see the typename replaced with auto.